spec/services/link_check_report/create_service_spec.rb

```diff
@@ -0,0 +1,217 @@
+require "spec_helper"
+
+RSpec.describe LinkCheckReport::CreateService do
+  let(:user) { FactoryGirl.create(:user) }
+
+  let(:link_check_report) { double(:link_check_report, valid?: nil, save!: nil) }
```

Comment:
Could we use a real object/factory here?

Reply:
I tried to use a factory here but because of the validation requirements it appears to be more effor than it's worth. I'm happy to get it working if it's required but I didn't feel like the time to get it working would be well spent

Reply:
Hmm, can we create a factory which satisfies the validation requirements?

Reply:
Yup, I ended up having to do this anyway for my PR so I'll copy that code over into here then rebase when this gets merged in